cmd/state-transition-update/main.go

```diff
@@ -59,9 +91,35 @@ func run() error {
 		return errs.Wrap(err, "Failed to check for latest update.")
 	}
 
-	err = up.InstallBlocking()
+	cfg, err := config.Get()
+	if err != nil {
+		return errs.Wrap(err, "Failed to read configuration.")
+	}
+	machineid.SetConfiguration(cfg)
+	machineid.SetErrorLogger(logging.Error)
+	logging.UpdateConfig(cfg)
+
+	if err := removeOldStateToolEnvironmentSettings(cfg); err != nil {
+		return errs.Wrap(err, "failed to remove environment settings from old State Tool installation")
+	}
+
+	err = up.InstallBlocking("")
 	if err != nil {
-		return errs.Wrap(err, "Failed to install mult-file update.")
+		return errs.Wrap(err, "Failed to install multi-file update.")
+	}
+
+	logging.Debug("Multi-file State Tool is installed.")
+
+	// if the transitional state tool has been replaced by the installer, we are done
+	stdout, _, err := exeutils.ExecSimple(appinfo.StateApp().Exec(), "__is_transitional")
+	if err != nil || strings.TrimSpace(stdout) != "true" {
+		return nil
+	}
```

Comment:
I'm not understanding what's happening here. Is the intend to detect that the executable has been replaced? Wouldn't the new state tool be installed elsewhere, and so this would only ever return the old?

Additionally, if you're just trying to check if a file has changed I think looking at the timestamp of the file would be a lot cleaner.

Reply:
Yes, if the State Tool was previously installed at the default installation path, this transitional tool is going to be replaced by the installer, and we do NOT want to remove it.

Checking the timestamp is an option, of course.  I will do that instead.

Reply:
But the default installation path is all new, there's no situation in which the old state tool would have been installed there unless someone manually set that path, which seems very unlikely.